juju/juju
# Use Base in Charmhub package and its response structures.

Make the least intrusive changes from platform to base while allowing for base to be used when communicating with the Charmhub API. This is a first step to replacement within the entire juju code base.

Add RefreshBase to correspond with RefreshPlatform.  All translation from RefreshPlatform to RefreshBase happens within the charmhub package.

Translate from a Base in the Charmhub API response structs inside of the charms and charmhub facades.

## QA steps

Check that the changes in charmhub/transport match the starcraft bases spec.

`juju info` and `juju find` will be tested once the appropriate API changes are ready for those calls. `juju info --series focal ubuntu-juju-qa` will not return correct data.  For now, ensure that the calls do not fail, but some data may be missing.

```console
$ juju bootstrap localhost  test --config logging-config="juju.apiserver.charms.client=TRACE; juju.apiserver.charmhub.client=TRACE; juju.resource.resourceadapters.charmhub.client=TRACE; juju.apiserver.charmrevisionupdater.charmhub=TRACE" --build-agent
$ juju add-model six --config charmhub-url="https://api.staging.snapcraft.io"

# ubuntu-juju-qa should deploy to bionic
$ juju deploy ubuntu-juju-qa
$ juju deploy ubuntu --series xenial

# There should be no errors related to installing the resource this config change initiates.
$ juju deploy postgresql --config aws_region="us-east-1"

# review the logs to see that base is used in charmhub api calls and responses.
$ juju debug-log -m controller
```
